nilearn/_utils/docs.py

```diff
@@ -262,7 +262,9 @@ def custom_function(vertices):
 ] = """
 data_dir : :obj:`pathlib.Path` or :obj:`str`, optional
```

Comment:
That is one thing I think we should probably discuss a bit because in many cases if the default is None then in many cases the code itself will set it to something else if it is None.

In this case if None is passed then data will be installed in the home folder tof the user.

There are many other cases like this in the codebase.

For the moment in this PR I have tried to mostly just move the default that was already in the description to the end of the "type" section.

But it may be good to sort of decide if this default should just be what is in the function declaration or if it should try to show what the "effective" default will be in the cases when None is passed.

I have a slight preference for the latter because it makes the doc a bit more informative and does not just repeat the function définition.

Reply:
Good point, we can leave for now.
Though this also applies when an option is e.g. "auto" and it could be the case that the "effective" default of a parameter being left as None changes according to other parameters. But this is or should be addressed rather in the parameter description.

Reply:
it is not always the home folder because this can be controlled with environment variables as well. so I think it might be a bit long to explain on the first line of the docstring entry

Reply:
True in that case but may be valid for other case where the default is `None`